src/Umbraco.Infrastructure/Media/ImageSharpDimensionExtractor.cs

```diff
@@ -0,0 +1,35 @@
+using System.IO;
+using SixLabors.ImageSharp;
+using Umbraco.Cms.Core.Media;
+using Size = System.Drawing.Size;
+
+namespace Umbraco.Cms.Infrastructure.Media
+{
+    internal class ImageSharpDimensionExtractor : IImageDimensionExtractor
+    {
+        /// <summary>
+        /// Gets the dimensions of an image.
+        /// </summary>
+        /// <param name="stream">A stream containing the image bytes.</param>
+        /// <returns>
+        /// The dimension of the image.
+        /// </returns>
+        public Size? GetDimensions(Stream stream)
+        {
+            if (stream.CanRead && stream.CanSeek)
+            {
+                stream.Seek(0, SeekOrigin.Begin);
+            }
```

Comment:
Not sure, but here if we can't read and seek the stream, wouldn't it make more sense to just return null, instead of still passing it to `Image.Identify`, or is image sharp doing some wild voodoo magic to still extract the data?

Reply:
I noticed the `Image.Identify` didn't use the configuration that's registered in the DI (required if the configuration is altered with additional image formats, etc.):

https://github.com/umbraco/Umbraco-CMS/blob/061f1b3b879f88f327a2fb215aa937811be9cbcf/src/Umbraco.Infrastructure/DependencyInjection/UmbracoBuilder.CoreServices.cs#L185

ImageSharp already contains a setting to automatically seek to the beginning if you want: [`Configuration.ReadOrigin`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Configuration.html#SixLabors_ImageSharp_Configuration_ReadOrigin). The default configuration uses `ReadOrigin.Current`, but within the CMS we only use a newly created stream, so the position will already be at the beginning anyway.